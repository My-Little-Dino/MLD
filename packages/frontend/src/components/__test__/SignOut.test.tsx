import React from 'react';
import ReactDOM from 'react-dom';
import { fireEvent, render } from '@testing-library/react';

import { useWeb3React } from '@web3-react/core';
import SignOut, { Props } from '../SignOut';

jest.mock('@web3-react/core');

describe('Test SignOut component', () => {
	const testProps: Props = {
		account: '0xcfffa6ace98eb3464f77af3059389f8f2c18b6e9',
		closeModal: jest.fn(),
	};
	const component = <SignOut {...testProps} />;

	const setup = () => render(component);

	beforeEach(() => {
		(useWeb3React as jest.Mock).mockReturnValue({
			deactivate: jest.fn(),
		});
	});

	afterEach(jest.clearAllMocks);

	it('Should render component without crash', () => {
		const div = document.createElement('div');
		ReactDOM.render(component, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Should match snapshot', () => {
		const { container } = setup();
		expect(container).toMatchSnapshot();
	});

	it('Should call deactivate & close modal when click sign out', () => {
		const { getByText } = setup();

		const signOut = getByText('Sign out');
		fireEvent.click(signOut);

		expect(useWeb3React().deactivate).toHaveBeenCalled();
		expect(testProps.closeModal).toHaveBeenCalled();
	});
});
