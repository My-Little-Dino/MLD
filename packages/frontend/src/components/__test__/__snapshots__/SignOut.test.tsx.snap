// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SignOut component Should match snapshot 1`] = `
<div>
  <div
    class="sc-bdnxRM hpGzml"
  >
    <p>
      You are connected:
    </p>
    <span>
      0xcfffa6ace98eb3464f77af3059389f8f2c18b6e9
    </span>
    <div
      class="sc-gtsrHT btuWiw"
    >
      Sign out
    </div>
  </div>
</div>
`;
